components/ai_chat/resources/untrusted_conversation_frame/components/assistant_reasoning/index.tsx

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import Icon from '@brave/leo/react/icon'
+import ProgressRing from '@brave/leo/react/progressRing'
+import { getLocale } from '$web-common/locale'
+import CaretSVG from '../svg/caret'
+import styles from './style.module.scss'
+
+interface QuoteProps {
+  text: string
+  isReasoning: boolean
+}
+
+function AssistantReasoning(props: QuoteProps) {
+  const [showDetails, setShowDetails] = React.useState<boolean>(false)
+
+  const handleShowHideDetails = (e: React.MouseEvent) => {
+    e.preventDefault()
+    setShowDetails((prev) => !prev)
+  }
+
+  return (
+    <div className={styles.wrapper}>
+      <div className={styles.reasoningRow}>
+        {props.isReasoning ? (
+          <ProgressRing className={styles.progressRing} />
+        ) : (
+          <Icon
+            className={styles.icon}
+            name='check-circle-outline'
+          />
+        )}
+        <span className={styles.reasoningLabel}>
+          {props.isReasoning
+            ? getLocale('reasoningLabel')
+            : getLocale('reasoningCompleteLabel')}
+        </span>
+        <a
```

Comment:
Hey could this just be a button? Semantically its not a link

Reply:
Changed to a `<button>`